### DISCUSSION TITLE
synchronise parametrised fixtures

### DISCUSSION THREAD

**Discussion by pablo-esteban** [pytest-dev/pytest#8278] [Q&A]
Hi all,

is there a way of synchronising the value of parametrising fixtures? For example:

```python
class TestAdd:
    _expected = {
        "test_add[1-1]": 2,
        "test_add[1-2]": 3,
        "test_add[2-1]": 3,
        "test_add[2-2]": 4,
    }

    @pytest.fixture(params=[1, 2])
    def x(self, request):
        return request.param

    @pytest.fixture(params=[1, 2])
    def y(self, request):
        return request.param

    @pytest.fixture
    def expected(self, request):
        return self._expected[request.node.name]

    def test_add(self, x, y, expected):
        observed = x+y
        assert observed == expected
```

The example above is exactly what i want, _but_ it is a hack: it is undesirable that the keys in the `_expected` dictionary are coupled to the test function name. I could not find a way of knowing the id of the current request, is that possible?

**Comment by bluetech** [comment]
Do you mean you'd like a `request` attribute to provides only the `1-1` part, instead of the full `test_add[1-1]`?

Currently pytest does not provide this. I think @RonnyPfannschmidt *might* have some things in mind that could help with that, but not in current pytest (short of parsing the string itself).

**Comment by RonnyPfannschmidt** [comment]
For the particular example, just use Parametrize directly with 2 arguments

If your case is more complex please expand the example
